test/e2e/apimachinery/flowcontrol.go

```diff
@@ -113,11 +126,158 @@ var _ = SIGDescribe("[Feature:APIPriorityAndFairness] response header should pre
 		}
 	})
 
+	ginkgo.It("should ensure that requests can't be drowned out", func() {
+		flowSchemaNamePrefix := "e2e-testing-flowschema"
+		priorityLevelNamePrefix := "e2e-testing-prioritylevel"
+		loadDuration := 10 * time.Second
+		type client struct {
+			username              string
+			qps                   float64
+			priorityLevelName     string
+			concurrencyMultiplier float64
+			concurrency           int32
+			flowSchemaName        string
+			matchingPrecedence    int32
+			completedRequests     int32
+		}
+		clients := []client{
+			// "elephant" refers to a client that creates requests at a much higher
+			// QPS than its counter-part and well above its concurrency share limit.
+			// In contrast, the mouse stays under its concurrency shares.
+			// Additionally, the "elephant" client also has a higher matching
+			// precedence for its flow schema.
+			{username: "elephant", qps: 100.0, concurrencyMultiplier: 2.0, matchingPrecedence: 999},
+			{username: "mouse", qps: 5.0, concurrencyMultiplier: 0.5, matchingPrecedence: 1000},
+		}
+
+		ginkgo.By("creating test priority levels and flow schemas")
+		for i := range clients {
+			clients[i].priorityLevelName = fmt.Sprintf("%s-%s", priorityLevelNamePrefix, clients[i].username)
+			framework.Logf("creating PriorityLevel %q", clients[i].priorityLevelName)
+			_, err := f.ClientSet.FlowcontrolV1alpha1().PriorityLevelConfigurations().Create(
```

Comment:
Sorry, I'm just now realizing that this makes separate priority levels for each. Usually we've been saying "mouse" and "elephant" to indicate users in the same priority level that we want to be fair to. If you're making separate priority levels, you're not really exercising the shuffle sharding.

Reply:
this tests the scenario described in the subsection titled "Priority" under request drownout here: https://github.com/kubernetes/kubernetes/issues/93757

I do plan to add another test to test fairness within a single priority level later as well.

Reply:
ah, perfect. "elephant" and "mouse" are confusing names in this context and would make more sense in the other. Can you fix in a followup?

Reply:
Will do :+1: